### DISCUSSION TITLE
setting which key for different mode

### DISCUSSION THREAD

**Discussion by davidgao7** [LazyVim/LazyVim#3589] [Q&A] [answered]
Hello everyone,

The partial text from original description for nvim-surround will be covered, so I want to modify the key description using which-key,

but I don't konw how to configure to show key hints for different mode (insert mode in this case),

I specified the mode to insert mode,  I've saved and reopen neovim(v0.10.0), and then I press <control+g> in insert mode, which-key is displaying but with the default description instead of the one I changed. 

<img width="846" alt="Screenshot 2024-06-11 at 11 10 57 AM" src="https://github.com/LazyVim/LazyVim/assets/39502459/f5c80d21-bd52-44a9-8e06-897b2306d9fd">

**Comment by dpetka2001** [comment, marked as answer]
I believe that's because the `defaults` table in LazyVim only has `{ "n", "v" }` mode. So, your changes in the defaults don't take effect. Just use `require("which-key").register` instead to see if that works. And with some latest changes I think you can also use `vim.keymap.set` to set group name descriptions. That could also be a choice for you to test out.
